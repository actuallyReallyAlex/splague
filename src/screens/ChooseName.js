import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Box, Button, FormField, TextInput } from 'grommet'
import {
  chooseName,
  changeScreen,
  transitionScreen
} from '../redux/actions/actions'

export class ChooseName extends Component {
  static propTypes = {
    // From mapStateToProps()
    cure: PropTypes.object.isRequired,
    // From connect()
    dispatch: PropTypes.func.isRequired,
    // From mapStateToProps()
    log: PropTypes.array.isRequired,
    // From mapStateToProps()
    plague: PropTypes.object.isRequired,
    // From mapStateToProps()
    player: PropTypes.object.isRequired,
    // From mapStateToProps()
    ui: PropTypes.object.isRequired,
    // From mapStateToProps()
    world: PropTypes.object.isRequired
  }

  state = {
    error: false,
    name: ''
  }

  playerName = React.createRef()

  handleNameSelection = e => {
    e.preventDefault()
    const { error } = this.state

    if (!error) {
      const { dispatch } = this.props
      const { name } = this.state
      dispatch(chooseName(name))
      dispatch(transitionScreen(true, 'animated fadeOut'))

      setTimeout(() => {
        dispatch(transitionScreen(false, 'animated fadeIn'))
        dispatch(changeScreen('chooseType'))
      }, 1500)
    }
  }

  handleNameChange = e => {
    e.preventDefault()
    const name = e.target.value
    const regex = /\w/gm
    if (regex.test(name)) {
      this.setState(() => ({ name, error: false }))
    } else {
      this.setState(() => ({ error: true }))
    }
  }

  render() {
    const { error, name } = this.state
    const { ui, player } = this.props
    return (
      <Box
        align="center"
        background={player.morality === 'good' ? 'accent-1' : '#252839'}
        className={ui.transitionClasses}
        fill
        justify="center"
      >
        <form>
          <Box align="center">
            <Box>
              <FormField
                error={error && "Name can't be blank. Please try again."}
                label="Name"
              >
                <TextInput
                  onChange={this.handleNameChange}
                  ref={this.playerName}
                />
              </FormField>
            </Box>
            {name && (
              <Box
                className="animated fadeInUp"
                margin={{ top: '130px' }}
                style={{ position: 'absolute' }}
              >
                <Button
                  label="Continue"
                  onClick={this.handleNameSelection}
                  primary
                  type="button"
                />
              </Box>
            )}
          </Box>
        </form>
      </Box>
    )
  }
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(ChooseName)
