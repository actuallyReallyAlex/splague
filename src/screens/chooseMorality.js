import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Box, RadioButton } from 'grommet'
import { chooseMorality } from '../redux/actions/actions'

class ChooseMorality extends Component {
  handleMoralitySelection = e => {
    const { dispatch } = this.props
    dispatch(chooseMorality(e.target.name))
  }

  render() {
    const { player } = this.props
    return (
      <Box align="center" fill justify="center">
        <Box direction="row" gap="medium">
          <RadioButton
            checked={player.morality === 'good'}
            label="Good"
            name="good"
            onChange={this.handleMoralitySelection}
          />
          <RadioButton
            checked={player.morality === 'evil'}
            label="Evil"
            name="evil"
            onChange={this.handleMoralitySelection}
          />
        </Box>
      </Box>
    )
  }
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(ChooseMorality)
