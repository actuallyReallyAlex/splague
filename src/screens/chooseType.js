import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Box, Button, FormField, Select } from 'grommet'
import { chooseType, changeScreen } from '../redux/actions/actions'
import { plagueTypes, factionTypes } from '../constants'

class ChooseType extends Component {
  state = {
    type: ''
  }

  handleBegin = () => {
    const { dispatch } = this.props
    const { type } = this.state
    dispatch(chooseType(type))
    dispatch(changeScreen('home'))
  }

  handleTypeChange = e => {
    const type = e.value
    this.setState(() => ({ type }))
  }

  render() {
    const { morality } = this.props.player
    const { type } = this.state

    return (
      <Box align="center" fill justify="center">
        <Box>
          <FormField label={morality === 'good' ? 'Faction' : 'Plague Type'}>
            <Select onChange={this.handleTypeChange} options={morality === 'good' ? factionTypes : plagueTypes} value={type} />
          </FormField>
        </Box>
        {type && (
          <Box margin={{ top: 'xlarge' }} style={{ position: 'absolute' }}>
            <Button label="Begin ..." onClick={this.handleBegin} primary />
          </Box>
        )}
      </Box>
    )
  }
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(ChooseType)
