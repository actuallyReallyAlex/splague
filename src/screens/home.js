import React from 'react'
import { connect } from 'react-redux'
import { Box } from 'grommet'
import Sidebar from '../components/Sidebar'

const Home = ({ player, ui }) => {
  return (
    <Box fill>
      <Sidebar />
    </Box>
  )
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(Home)
