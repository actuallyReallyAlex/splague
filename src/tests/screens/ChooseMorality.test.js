import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import 'jest-styled-components'
import { ChooseMorality } from '../../screens/ChooseMorality'
import developmentState from '../fixtures/developmentState'

const dispatch = jest.fn()

describe('<ChooseMorality />', () => {
  it('Should render snapshot of ChooseMorality screen.', () => {
    const { cure, log, plague, player, ui, world } = developmentState
    const renderer = new ShallowRenderer()
    renderer.render(
      <ChooseMorality
        cure={cure}
        dispatch={dispatch}
        log={log}
        plague={plague}
        player={player}
        ui={ui}
        world={world}
      />
    )
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
