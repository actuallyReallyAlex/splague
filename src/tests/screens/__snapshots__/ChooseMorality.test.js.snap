// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseMorality /> Should render snapshot of ChooseMorality screen. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-right: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  height: 24px;
  width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px rgba(0,0,0,0.15);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 100%;
}

.c10 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
}

.c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c4:hover input:not([disabled]) + div,
.c4:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c8 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  margin: 0;
  z-index: 1;
  cursor: pointer;
}

.c6 {
  position: relative;
}

@media only screen and (max-width:768px) {
  .c0 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    border: solid 2px rgba(0,0,0,0.15);
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    padding: 0px;
  }
}

<div
  className="animated fadeIn slow c0"
>
  <form>
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <div
          className="c3"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <label
            className="c4 c5"
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="c6 c7"
            >
              <input
                checked={false}
                className="c8"
                disabled={false}
                name="good"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
              />
              <div
                className="c9"
              />
            </div>
            <span>
              Good
            </span>
          </label>
        </div>
        <div
          className="c10"
        />
        <div
          className="c3"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <label
            className="c4 c5"
            disabled={false}
            onClick={[Function]}
          >
            <div
              className="c6 c7"
            >
              <input
                checked={false}
                className="c8"
                disabled={false}
                name="evil"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
              />
              <div
                className="c9"
              />
            </div>
            <span>
              Evil
            </span>
          </label>
        </div>
      </div>
    </div>
  </form>
</div>
`;
